src/FavoritesContext.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useContext, useEffect} from 'react'
+
+
+const FavoritesContext = React.createContext(null)
+
+function useFavorites() {
+    const context = useContext(FavoritesContext);
+    if (!context) {
+      throw new Error(`Can't use "useSearch" without an Favorites!`);
+    }
+    return context;
+  }
+
+  function FavoritesProvider({ children }) {
+    const [favorites, setFavorites] = useState([])
+
+    function getFavorites() {
+        let favorites=JSON.parse(sessionStorage.getItem('favorites'))
+
+        if (!favorites) {
+            favorites = []
+        } 
+        setFavorites(favorites)
+    }
+
+    function removeFavorite(video) {
+
+        const fav = favorites.filter((e)=>{
+            return e.id.videoId !== video.id.videoId
+        })
+        saveFavorites(fav)
+      
+
+    }
+
+    function addFavorite(video) {
+
+        let idx = favorites.findIndex(x => x.id.videoId ===video.id.videoId)
+        console.log(idx)
+
+        if (idx===-1)
+        {
+            const fav = [...favorites, video]
+            saveFavorites(fav)
+        }
+    }
+
+    function saveFavorites(favorites) {
+
+        sessionStorage.setItem('favorites', JSON.stringify(favorites));
+        setFavorites(favorites)
+
+    }
+
+    useEffect(() => {
+        getFavorites()
+      }, [])
+
+    return (
+      <FavoritesContext.Provider value={{ favorites, addFavorite, removeFavorite}}>
+        {children}
+      </FavoritesContext.Provider>
+    );
+  }
+
+
+export { useFavorites};
+export default FavoritesProvider;
```

Comment:
renderProps + Context + customHook 🤯  My brain still hurts from trying to understand this.

From your current implementation the only component that needs to know the list of Favorites is **FavoriteVideos** so I recommend you put the state directly there. Also the **getFavorites()**, **addFavorite(favorite)** and **removeFavorite(id)** will most likely evolve into an api that will connect to a long term storage, so I recommend creating a Storage module with these methods in it.

e.g.

```js
---- FavoriteVideos.js ---
import { getFavorites, removeFavorite } from './storage';

function FavoriteVideos(props) {
    const [favorites, setFavorites] = useState([]);
    
    sideEffect(() => {
        setFavorites(getFavorites());
    }, [])
   .
   .
   .
   <FavoriteVideoCard
            key= {video.id.videoId}
            video={video}
            removeFavorite={() => {
                 removeFavorite(video.id);
                 ... update state
            }}
    />
}


---- Videos.js ---
import { addFavorite } from './storage';

function Videos(props) {
.
.
.
     <Video
          addFavorite = {() => addFavorite(video)}
      >
}

```